### PR TITLE
Add column encryption_key to table alicloud_rds_instance closes #296

### LINKED ISSUE #296: Add column `encryption_key` to table `alicloud_rds_instance`
Instance API - https://www.alibabacloud.com/help/en/apsaradb-for-rds/latest/query-instances
API for Security group - DescribeSecurityGroupConfiguration

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/alicloud_rds_instance []

PRETEST: tests/alicloud_rds_instance

TEST: tests/alicloud_rds_instance
Running terraform
data.alicloud_zones.resource: Reading...
data.alicloud_caller_identity.current: Reading...
data.alicloud_caller_identity.current: Read complete after 1s [id=295372922530649815]
data.null_data_source.resource: Reading...
data.null_data_source.resource: Read complete after 0s [id=static]
data.alicloud_zones.resource: Read complete after 4s [id=831125506]

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # alicloud_db_instance.named_test_resource will be created
  + resource "alicloud_db_instance" "named_test_resource" {
      + acl                        = (known after apply)
      + auto_upgrade_minor_version = (known after apply)
      + babelfish_port             = (known after apply)
      + ca_type                    = (known after apply)
      + category                   = (known after apply)
      + connection_string          = (known after apply)
      + connection_string_prefix   = (known after apply)
      + db_instance_storage_type   = (known after apply)
      + db_is_ignore_case          = (known after apply)
      + db_time_zone               = (known after apply)
      + deletion_protection        = false
      + engine                     = "MySQL"
      + engine_version             = "5.7"
      + force_restart              = false
      + ha_config                  = (known after apply)
      + id                         = (known after apply)
      + instance_charge_type       = "Postpaid"
      + instance_name              = "turbottest18968"
      + instance_storage           = 30
      + instance_type              = "rds.mysql.s2.large"
      + maintain_time              = (known after apply)
      + monitoring_period          = 60
      + port                       = (known after apply)
      + private_ip_address         = (known after apply)
      + replication_acl            = (known after apply)
      + resource_group_id          = (known after apply)
      + security_group_id          = (known after apply)
      + security_group_ids         = (known after apply)
      + security_ip_mode           = "normal"
      + security_ips               = (known after apply)
      + server_cert                = (known after apply)
      + server_key                 = (known after apply)
      + sql_collector_config_value = 30
      + sql_collector_status       = (known after apply)
      + ssl_action                 = (known after apply)
      + ssl_status                 = (known after apply)
      + target_minor_version       = (known after apply)
      + tcp_connection_type        = (known after apply)
      + vpc_id                     = (known after apply)
      + vswitch_id                 = (known after apply)
      + zone_id                    = (known after apply)
      + zone_id_slave_a            = (known after apply)

      + babelfish_config {
          + babelfish_enabled    = (known after apply)
          + master_user_password = (known after apply)
          + master_username      = (known after apply)
          + migration_mode       = (known after apply)
        }

      + parameters {
          + name  = (known after apply)
          + value = (known after apply)
        }
    }

  # alicloud_vpc.named_test_resource will be created
  + resource "alicloud_vpc" "named_test_resource" {
      + cidr_block            = "172.16.0.0/16"
      + id                    = (known after apply)
      + ipv6_cidr_block       = (known after apply)
      + name                  = (known after apply)
      + resource_group_id     = (known after apply)
      + route_table_id        = (known after apply)
      + router_id             = (known after apply)
      + router_table_id       = (known after apply)
      + secondary_cidr_blocks = (known after apply)
      + status                = (known after apply)
      + vpc_name              = "turbottest18968"
    }

  # alicloud_vswitch.named_test_resource will be created
  + resource "alicloud_vswitch" "named_test_resource" {
      + availability_zone = (known after apply)
      + cidr_block        = "172.16.0.0/24"
      + id                = (known after apply)
      + name              = (known after apply)
      + status            = (known after apply)
      + vpc_id            = (known after apply)
      + vswitch_name      = "turbottest18968"
      + zone_id           = "us-east-1a"
    }

Plan: 3 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + db_instance_id = (known after apply)
  + port           = (known after apply)
  + resource_aka   = (known after apply)
  + resource_name  = "turbottest18968"
  + zone_id        = (known after apply)
alicloud_vpc.named_test_resource: Creating...
alicloud_vpc.named_test_resource: Creation complete after 7s [id=vpc-0xiqmt5wetn17flj2buly]
alicloud_vswitch.named_test_resource: Creating...
alicloud_vswitch.named_test_resource: Creation complete after 8s [id=vsw-0xit7mdcfa35cmhu7nceb]
alicloud_db_instance.named_test_resource: Creating...
alicloud_db_instance.named_test_resource: Still creating... [10s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [20s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [30s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [40s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [50s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [1m0s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [1m10s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [1m20s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [1m30s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [1m40s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [1m50s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [2m0s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [2m10s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [2m20s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [2m30s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [2m40s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [2m50s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [3m0s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [3m10s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [3m20s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [3m30s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [3m40s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [3m50s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [4m0s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [4m10s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [4m20s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [4m30s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [4m40s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [4m50s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [5m0s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [5m10s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [5m20s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [5m30s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [5m40s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [5m50s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [6m0s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [6m10s elapsed]
alicloud_db_instance.named_test_resource: Still creating... [6m20s elapsed]
alicloud_db_instance.named_test_resource: Creation complete after 6m27s [id=rm-7go63vipj009bgu22]

Warning: Deprecated

  with data.null_data_source.resource,
  on variables.tf line 23, in data "null_data_source" "resource":
  23: data "null_data_source" "resource" {

The null_data_source was historically used to construct intermediate values
to re-use elsewhere in configuration, the same can now be achieved using
locals

(and one more similar warning elsewhere)

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

db_instance_id = "rm-7go63vipj009bgu22"
port = "3306"
resource_aka = "arn:acs:rds:us-east-1:123379499374298:instance/rm-7go63vipj009bgu22"
resource_name = "turbottest18968"
zone_id = "us-east-1a"

Running SQL query: test-get-query.sql
[
  {
    "db_instance_id": "rm-7go63vipj009bgu22",
    "db_instance_type": "Primary",
    "engine": "MySQL",
    "engine_version": "5.7"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "arn": "arn:acs:rds:us-east-1:123379499374298:instance/rm-7go63vipj009bgu22",
    "port": "3306"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "db_instance_storage": 30,
    "zone_id": "us-east-1a"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:acs:rds:us-east-1:123379499374298:instance/rm-7go63vipj009bgu22"
    ],
    "title": "rm-7go63vipj009bgu22"
  }
]
✔ PASSED

POSTTEST: tests/alicloud_rds_instance

TEARDOWN: tests/alicloud_rds_instance

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select 
  i.arn as instance_arn,
  i.title as instance_name,
  encryption_key,
  k.title as kms_key_name
from 
  alicloud_rds_instance i 
  left join alicloud_kms_key k 
    on encryption_key = key_id
+------------------------------------------------------------------------+----------------------+--------------------------------------+---------------+
| instance_arn                                                           | instance_name        | encryption_key                       | kms_key_name  |
+------------------------------------------------------------------------+----------------------+--------------------------------------+---------------+
| arn:acs:rds:cn-hangzhou:1232111499156037:instance/rm-1uda02co12gd0374y | rm-1uda02co12gd0374y | 86f0f18e-8e99-478c-8cba-8464f891a6cb | alias/test123 |
+------------------------------------------------------------------------+----------------------+--------------------------------------+---------------+
```
</details>

### CODE REVIEW

File: docs/tables/alicloud_rds_instance.md [hunk 1]
```diff
@@ -103,4 +100,17 @@ from
...
   tde_status = 'Disabled';
+```
+
+### Get security group configuration details for instances
+
+```sql
+select
+  i.db_instance_id,
+  s ->> 'NetworkType' as network_type,
+  s ->> 'RegionId' as region_id,
```
Comment (misraved) [inline docs/tables/alicloud_rds_instance.md:111]:
```suggestion
  s ->> 'RegionId' as security_group_region_id,
```

**Review by misraved — COMMENTED** [review]
@ParthaI please take a look at the review comments. 
Does the `encryption_key` column represent only the encryption key set by the users and not the ones that are enabled by Alicloud?

**Comment by Priyanka-Chatterjee-2000** [comment]
This column is always coming null even after querying an encryption enabled rds instance. Please can you check @ParthaI !! @madhushreeray30 and @karanpopat was also facing the same issue
`select encryption_key from alicloud_rds_instance
+----------------+
| encryption_key |
+----------------+
| <null>         |
| <null>         |
+----------------+`

**Comment by ParthaI** [comment]
> This column **encryption_key** is always coming null even after querying an encryption enabled rds instance. Please can you check @ParthaI !! @madhushreeray30 and @karanpopat was also facing the same issue
> 
> ```
> select encryption_key from alicloud_rds_instance
> +----------------+
> | encryption_key |
> +----------------+
> | <null>         |
> | <null>         |
> +----------------+
> ```

Unable to test is again because, we are not able to create resource due to some console error.
I will update this PR once the resource is being created.
Thanks!

**Comment by misraved** [comment]
@ParthaI @karanpopat did we manage to check if the data issue still persists?

**Comment by karanpopat** [comment]
@misraved we are not able to create an RDS instance like before. Have raised a support ticket with the alicloud team. Will update once we have a resolution

**Review by misraved — APPROVED** [review]

**Pull request merged**